drivers/FlashIAP.cpp

```diff
@@ -56,8 +58,12 @@ int FlashIAP::init()
 {
     int ret = 0;
     _mutex->lock();
-    if (flash_init(&_flash)) {
-        ret = -1;
+    {
+        ScopedMpuXnLock xn;
+        ScopedMpuWnLock wn;
```

Comment:
Maybe we're slightly on the wrong track by "locking out" something, as this and the "deep sleep lock" do, rather than "locking in/on" something. By contrast Android's auto-suspend is controlled by a "wake lock", not a "suspend lock". Still, given the deep sleep lock precedent, may as well stick with this.

Reply:
`CriticalSectionLock` has the opposite naming as it keep you in a critical section. Would names like `ScopedRamExecutionLock` and `ScopedRomWriteLock` be preferred going forward?